calico.py

```diff
@@ -125,6 +125,19 @@ def __init__(self, network_config, env):
         Path in which to search for CNI plugins.
         """
 
+        if self.k8s_namespace and self.k8s_pod_name:
+            self.workload_id = "%s.%s" % (self.k8s_namespace, self.k8s_pod_name)
```

Comment:
What about adding a `self.running_under_k8s = True` line. Then elsewhere, you get more readable code by just checking that(e.g. line 178)